packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/gollm/GoLLMController.java

```diff
@@ -73,12 +69,16 @@ private TaskResponseHandler getModelCardResponseHandler() {
 			try {
 				final String serializedString = objectMapper.writeValueAsString(resp.getAdditionalProperties());
 				final ModelCardProperties props = objectMapper.readValue(serializedString, ModelCardProperties.class);
-				log.info("Writing model card to database for model {}", props.getModelId());
-				final Model model = modelService.getAsset(props.getModelId())
+				log.info("Writing model card to database for document {}", props.getDocumentId());
+				final DocumentAsset document = documentAssetService.getAsset(props.getDocumentId())
 						.orElseThrow();
 				final ModelCardResponse card = objectMapper.readValue(resp.getOutput(), ModelCardResponse.class);
-				model.getMetadata().setGollmCard(card.response);
-				modelService.updateAsset(model);
+				if (document.getMetadata() == null){
+					document.setMetadata(new java.util.HashMap<>());
+				}
+				document.getMetadata().put("goLLM_card", card.response);
```

Comment:
We agreed that `snake_case` are ~the devil~ not to be used anyware in the HMI.
For JSON properties we use `camelCase`, so in this case:
```suggestion
				document.getMetadata().put("gollmCard", card.response);
```

Reply:
changed all instances to `gollmCard`